### DISCUSSION TITLE
Using es modules in node with workspaces

### DISCUSSION THREAD

**Discussion by wdfinch** [yarnpkg/berry#2891] [Q&A] [answered]
Hi all! This is probably something super simple, but I can't figure this out after a few hours. Would really appreciate it if someone could point me in the right direction.

I have two workspaces. One exports a string constant. The other imports it and prints it. I'm using es modules to do this over common js. No matter what I try, I keep getting `Error [ERR_MODULE_NOT_FOUND]`. 

I've tried adding `type: module` to both package.jsons and adding the `.mjs` extension. I'm using yarn node to run the script. My node version is `15.14.0`.

Can anyone please explain how I can use es modules with workspaces for node js scripts?

Note es modules work fine for non-workspace (relative) imports and common js does work for workspace imports.

I've attached my sample project below.

[workspaces-es-modules-node.zip](https://github.com/yarnpkg/berry/files/6475868/workspaces-es-modules-node.zip)

**Comment by merceyz** [comment, marked as answer]
Yarn PnP doesn't support ESM yet (https://github.com/yarnpkg/berry/issues/638), I implemented it in https://github.com/yarnpkg/berry/pull/2161 but due to the state of loaders in Node I've left it alone in a draft state. Last time I tested it it did work so if you want to give it a try feel free to do so.

```
yarn set version from sources --branch 2161
yarn config set enableExperimentalESMLoader true
yarn
```

**Reply by wdfinch** [reply]
Gotcha. Thanks so much for the quick reply!
